lib/utils/replace.go

```diff
@@ -46,5 +46,26 @@ func ReplaceRegexp(expression string, replaceWith string, input string) (string,
 	return expr.ReplaceAllString(input, replaceWith), nil
 }
 
+func SliceContainsRegexStr(input string, expressions []string) (bool, error) {
+	for _, expression := range expressions {
+		if !strings.HasPrefix(expression, "^") || !strings.HasSuffix(expression, "$") {
+			// replace glob-style wildcards with regexp wildcards
+			// for plain strings, and quote all characters that could
+			// be interpreted in regular expression
+			expression = "^" + GlobToRegexp(expression) + "$"
+		}
+
+		expr, err := regexp.Compile(expression)
+		if err != nil {
+			return false, trace.BadParameter(err.Error())
+		}
+		if expr.MatchString(input) {
```

Comment:
Add a comment that because the expression is always surrounded by `^` and `$` this is an exact match for either a a plain string (for example `^hello$`) or for a regexp (for example `^hel*o$`).